### DISCUSSION TITLE
Learning game with vexflow?

### DISCUSSION THREAD

**Discussion by willkessler** [0xfe/vexflow#1550] [Q&A]
Hello, I'm new to vexflow and it's pretty amazing.

I'm also a beginner bass guitar player. To teach myself, I thought I'd write a simple game so I can learn the bass clef, the fretboard, etc, in a fun way.  (I was trained as a kid with treble clef so I keep getting mixed up.)

Does anybody know if there are any learning games already in existence based on vexflow out there I could check out? Cheers!

**Comment by willkessler** [comment]
Actually, although it didn't come up in a search for game, looking around at other posts I did find uTheory which kind of gets at what I'd like to learn with eg. [here](https://utheory.com/app/lessons/bass_clef_2/bass_clef_2_play_low_ledgers) 
But what I want to do is play along as well on my bass guitar to make my brain associate fretboard with notes on the scale, and eventually learn to play sequences like scales just by sight-reading.

**Comment by willkessler** [comment]
I also found a few other "games" just now like this one at [Musicca](https://www.musicca.com/exercise/36) . Maybe they use vexflow as well. hard to tell. But I'm dumb and these games are frustrating to learn with, sort of just brute force. As soon as I learn 3 notes and try the next three notes, then I forget the first three.... I really need something that ties the guitar to the written music more physically vs clicking a mouse.

**Comment by rvilarl** [comment]
I wrote pianoplay to learn reading scores for piano. It should be easy to adapt it to other instruments with midi interface.

https://github.com/rvilarl/pianoplay

if you have a keyboard with mid interface it will already help you to read scores. It uses VexFlow.
